src/Discovery/Discovery.tsx

```diff
@@ -0,0 +1,552 @@
+                }}
+              >
+                {renderValue(studyPreviewText)}
+              </div>
+            );
+          },
+          expandedRowClassName: () => 'discovery-table__expanded-row',
+          expandIconColumnIndex: -1, // don't render expand icon
+        })}
+      />
+    </div>
+    <Modal
+      className='discovery-modal'
+      visible={modalVisible}
+      onOk={() => setModalVisible(false)}
+      onCancel={() => setModalVisible(false)}
+      width='80vw'
+      footer={false}
+    >
+      <Space direction='vertical' size='large'>
+        { config.study_page_fields.header &&
+          <Space align='baseline'>
+            <h3 className='discovery-modal__header-text'>{modalData[config.study_page_fields.header.field]}</h3>
+            <a href={`/discovery/${modalData[config.minimal_field_mapping.uid]}/`}><LinkOutlined /> Permalink</a>
+          </Space>
+        }
+        { config.features.authorization.enabled &&
+          (modalData[accessibleFieldName]
+            ? (
+              <Alert
+                className='discovery-modal__access-alert'
+                type='success'
+                message={<><UnlockOutlined /> You have access to this study.</>}
+              />
+            )
+            : (
+              <Alert
+                className='discovery-modal__access-alert'
+                type='warning'
+                message={<><LockFilled /> You do not have access to this study.</>}
+              />
+            )
+          )
+        }
+        { config.study_page_fields.fields_to_show.map((fieldGroup, i) => (
+          <div key={i} className='discovery-modal__attribute-group'>
+            { fieldGroup.include_name &&
+                  <h3 className='discovery-modal__attribute-group-name'>{fieldGroup.group_name}</h3>
+            }
+            { fieldGroup.fields.map((field) => {
+              // display nothing if selected study doesn't have this field
+              // and this field isn't configured to show a default value
+              if (!modalData[field.field] && !field.include_if_not_available) {
+                return null;
+              }
+              return (
+                <div key={field.name} className='discovery-modal__attribute'>
+                  { field.include_name !== false &&
+                        <span className='discovery-modal__attribute-name'>{field.name}</span>
+                  }
+                  <span className='discovery-modal__attribute-value'>
+                    { modalData[field.field]
+                      ? renderFieldContent(modalData[field.field], field.content_type)
+                      : (field.value_if_not_available || 'Not available')
+                    }
+                  </span>
+                </div>
+              );
+            })}
+          </div>
+        ))}
+      </Space>
+    </Modal>
+  </div>);
+};
+
+Discovery.defaultProps = {
+  params: { studyUID: undefined },
```

Comment:
I'm not 100% sure, but I think that `null` would be a better initializer here. Null is used to define something programmatically empty. Undefined is meant to say that the reference is not existing. If I would make a property intentionally empty but still present, I would use null rather than undefined. Paraphrased from https://stackoverflow.com/questions/6604749/what-reason-is-there-to-use-null-instead-of-undefined-in-javascript?rq=1